src/internal/utils/file_utils.go

```
@@ -37,66 +38,74 @@
+	// Check for missing fields
+	targetType := reflect.TypeOf(target).Elem()
+	missingFields := []string{}
 
-		for i := range targetType.NumField() {
-			field := targetType.Field(i)
-			if _, exists := rawData[field.Tag.Get("toml")]; !exists {
-				hasError = true
-				// A field doesn't exist in the toml config file
-				errMsg += errorPrefix + fmt.Sprintf("Field \"%s\" is missing\n", field.Tag.Get("toml"))
-			}
+	for i := 0; i < targetType.NumField(); i++ {
```

golangci-lint (1.64.8)

for loop can be changed to use an integer range (Go 1.22+)

(intrange)